### PR TITLE
fix the wifi configuration some more

### PR DESCRIPTION
**Pull request by mwhudson**

### CODE REVIEW

File: subiquitycore/ui/views/network_configure_wlan_interface.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def done(self, sender):
         if self.device.configured_ssid[0] is None and self.form.ssid.value:
             # Turn DHCP4 on by default when specifying an SSID for
             # the first time...
-            self.device.dhcp4 = True
+            self.device.config['dhcp4'] = True
```
Comment (xnox) [inline subiquitycore/ui/views/network_configure_wlan_interface.py:154]: Hm, should we enable dhcp6 too here?

Most of IoT is ipv4 still, but it should at least try ipv6 in 2020, right?

Comment (waveform80) [reply subiquitycore/ui/views/network_configure_wlan_interface.py:154]: A brief data-point on this: Mythic's Pi cloud is entirely IPv6 only (with a proxy to provide web-access to IPv4-only clients) so I'd say it's certainly getting more important in the IoT space.

**Comment by vorlonofportland** [comment]
On Tue, May 19, 2020 at 06:07:44AM -0700, Dimitri John Ledkov wrote:
> @xnox commented on this pull request.
> 
> 
> 
> > @@ -151,7 +151,7 @@ def done(self, sender):
>          if self.device.configured_ssid[0] is None and self.form.ssid.value:
>              # Turn DHCP4 on by default when specifying an SSID for
>              # the first time...
> -            self.device.dhcp4 = True
> +            self.device.config['dhcp4'] = True

> Hm, should we enable dhcp6 too here?

> Most of IoT is ipv4 still, but it should at least try ipv6 in 2020, right?

Most ipv6 networking is SLAAC, not DHCPv6; unless you're in an ipv6-only
environment, you probably don't need DHCPv6.

**Comment by xnox** [comment]
> On Tue, May 19, 2020 at 06:07:44AM -0700, Dimitri John Ledkov wrote: @xnox commented on this pull request. > @@ -151,7 +151,7 @@ def done(self, sender): if self.device.configured_ssid[0] is None and self.form.ssid.value: # Turn DHCP4 on by default when specifying an SSID for # the first time... - self.device.dhcp4 = True + self.device.config['dhcp4'] = True Hm, should we enable dhcp6 too here? Most of IoT is ipv4 still, but it should at least try ipv6 in 2020, right?
> Most ipv6 networking is SLAAC, not DHCPv6; unless you're in an ipv6-only environment, you probably don't need DHCPv6.

right but unless netplan's dhcp6 is set, there is no SLAAC, either right?

**Comment by vorlonofportland** [comment]
On Tue, May 19, 2020 at 12:12:11PM -0700, Dimitri John Ledkov wrote:

> right but unless netplan's dhcp6 is set, there is no SLAAC, either right?

There's no reason this should be the case.  That is certainly not the design
of netplan.

**Comment by mwhudson** [comment]
We do need to straighten out the IPv6 story a bit, see recent subiquity bug reports about that, but AIUI we should not be enabling DHCPv6 by default.

**Comment by mwhudson** [comment]
PS: thanks for merging :)

**Pull request merged**